tests/posix/fmemopen.c

```diff
@@ -0,0 +1,119 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define BUFFER0_SIZE 0x1000
+#define BUFFER1 "Hello world"
+#define BUFFER1_SIZE (sizeof(BUFFER1))
+
+int main() {
+	// test seek with mode "r"
+	FILE *f = fmemopen(NULL, BUFFER0_SIZE, "r");
+	assert(f);
+
+	int ret = fseek(f, 0, SEEK_END);
+	assert(!ret);
+	long pos = ftell(f);
+	fprintf(stderr, "pos = %ld\n", pos);
+	// Despite the fact that this is correct behavior (see below),
+	// this sometimes fails on glibc; newlibs seems to get this wrong, too.
+	// to quote what posix says about it:
+	// > The stream shall also maintain the size of the current buffer contents;
+	// > use of fseek() or fseeko() on the stream with SEEK_END shall seek relative to this size.
+	// > For modes r and r+ the size shall be set to the value given by the size argument.
+#if !defined(__GLIBC__)
+	assert(pos == BUFFER0_SIZE);
+#endif
+
+	fclose(f);
+
+	// test seek with mode "w"
+	f = fmemopen(NULL, BUFFER0_SIZE, "w");
```

Comment:
Missing test case: testing the initial conditions of the buffer (the implementation should place a null byte at the first byte).

Reply:
Yeah that's fair. I think we don't have to test for this.